File: ruby/abc194_c.rb
n, *a = $<.read.split.map &:to_i
b = [0]*401
a.each do |x|
  b[x+200] += 1
end
ans = 0
(0..400).each do |x|
  (x..400).each do |y|
    ans += b[x]*b[y]*(x-y)**2
  end
end
puts ans

File: csharp/abc194_c.cs
using System;
using System.Linq;
class ABC
{
  static void Main()
  {
    long ans = 0;
    Console.ReadLine();
    var x = Console.ReadLine().Split(" ").Select(int.Parse).ToArray();
    var a = new long[401];
    for(var i=0; i<x.Length; i++) a[x[i]+200]++;
    for(var i=0; i<400; i++)
    {
      for(var j=i+1; j<401; j++)
      {
        ans += (long)(a[i]*a[j]*(i-j)*(i-j));
      }
    }
    Console.WriteLine(ans);
  }
}